scss/_patterns_navigation.scss

```diff
@@ -147,230 +146,204 @@
+      justify-content: space-between;
+      width: 100%;
+    }
+  }
 
-      &:last-of-type {
-        border-right: 1px solid $color-mid-light;
+  .p-search-box {
+    min-width: 10em;
 
-        @media (max-width: $breakpoint-navigation-threshold) {
-          border-bottom: 0;
-          border-right: 0;
-        }
-      }
+    @media (max-width: $breakpoint-navigation-threshold) {
+      flex: 1 0 auto;
+      margin: -1px $grid-margin-width $spv-intra $grid-margin-width;
+      order: -1;
     }
 
-    &__nav {
-      display: none;
-      margin-top: 0;
-
-      @media (min-width: $breakpoint-navigation-threshold + 1) {
-        display: flex;
-      }
+    @media (min-width: $breakpoint-navigation-threshold + 1) {
+      // align baselines of menu items and input text
+      $input-gap-top: $spv-intra--expanded - $spv-nudge;
+      display: flex;
+      flex: 1 1 auto;
+      margin: $input-gap-top $sph-intra auto auto;
+      max-width: 20rem;
+      order: 1;
+    }
+  }
 
-      > .p-search-box {
-        align-self: center;
-        margin-left: auto;
-        margin-top: 0;
-        max-width: 20rem;
-        order: 1;
-        width: 100%;
-
-        @media (max-width: $breakpoint-navigation-threshold) {
-          max-width: inherit;
-          order: -1;
-          top: 0;
-        }
+  &__row {
+    display: flex;
+    // XXX: these zero out the default row padding to avoid doubling.
+    // I think a more robust soljution would be to wrap the nav in a row, rather than nest one ins it and override it.
+    padding-left: 0;
+    padding-right: 0;
+    width: 100%;
 
-        > .p-search-box__input,
-        .p-search-box__button,
-        .p-search-box__reset {
-          @media (max-width: $breakpoint-navigation-threshold) {
-            border-bottom: 0;
-          }
-        }
-      }
+    @media (max-width: $breakpoint-navigation-threshold) {
+      flex-direction: column;
     }
   }
 
   &:target {
+    .p-navigation__nav {
+      display: flex;
+    }
+
     .p-navigation__toggle--open {
       display: none;
     }
 
     .p-navigation__toggle--close {
       @media (max-width: $breakpoint-navigation-threshold) {
-        display: inline-block;
+        display: block;
       }
     }
+  }
 
-    .p-navigation__nav {
-      display: flex;
-      flex-direction: row;
+  &__toggle {
+    &--open,
+    &--close {
+      display: none;
+      margin: 0 $grid-margin-width auto $sph-intra;
+      padding: $spv-intra--expanded 0;
+    }
 
+    &--open {
       @media (max-width: $breakpoint-navigation-threshold) {
-        flex-direction: column;
+        display: block;
       }
     }
   }
+
+  .u-image-position .u-image-position--right {
+    order: 2; // ensure it is to the right of search
+    // XXX: patch - override utility, as it positions absolutely, which leads to overlap on smaller screens.
+    // Should use flex instead; But that will require changing the markup
+    position: relative;
+    right: unset;
+  }
```

Comment:
I think we should leave the image-position as it is and create something for this pattern to align right.

Reply:
leaving as is for now as doing it otherwise would break things that use the current markup.